packages/react-components/react-list-preview/src/components/List/List.types.ts

```diff
@@ -1,17 +1,57 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import type {
+  ComponentProps,
+  ComponentState,
+  OnSelectionChangeData,
+  Slot,
+  SelectionMode,
+  SelectionItemId,
+} from '@fluentui/react-utilities';
+import type { UseArrowNavigationGroupOptions } from '@fluentui/react-components';
+import { ListSelectionState } from '../../hooks/types';
+
+export enum ListLayout {
+  Horizontal = 'horizontal',
+  Vertical = 'vertical',
+  Grid = 'grid',
+}
+
+// Component ref interface
+export type IList = {
+  selection: ListState['selection'];
+};
 
 export type ListSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'ul', 'div' | 'ol' | 'dl'>>;
 };
 
 /**
  * List Props
  */
-export type ListProps = ComponentProps<ListSlots> & {};
+export type ListProps = ComponentProps<ListSlots> & {
+  layout?: ListLayout;
+  customArrowNavigationOptions?: Partial<UseArrowNavigationGroupOptions>;
+  focusableItems?: boolean;
+  selectable?: boolean;
+  selectionMode?: SelectionMode;
+  defaultSelectedItems?: SelectionItemId[];
+  onSelectionChange?: (event: React.SyntheticEvent, data: OnSelectionChangeData) => void;
+  componentRef?: React.Ref<IList>;
```

Comment:
removed, in case the user wants to use the nice selection methods, they will be able to use `useListSelection` hook to manage it. Otherwise, they can rely on a basic react controlled pattern using `React.setState`.